metrics/src/metrics.rs

```diff
@@ -515,7 +509,7 @@ mod test {
         }
 
         agent.flush();
-        assert_eq!(writer.points_written(), 20);
+        assert_eq!(writer.points_written(), 21);
```

Comment:
All these counts increase by one because I modified the `flush` implementation to re-use `Self::write()`, which internally always adds an additional datapoint to report on the performance of metrics itself